test/timer_test.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2018 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <unistd.h>
+#include <sstream>
+
+#include "gtest/gtest.h"
+#include "source/util/timer.h"
+
+namespace {
+
+using ::spvutils::CumulativeTimer;
+using ::spvutils::PrintTimerDescription;
+using ::spvutils::ScopedTimer;
+using ::spvutils::Timer;
+
+TEST(Timer, Sleep) {
+  const unsigned int usleep_time = 71234;
+  const double epsilon = 0.001;
+  std::ostringstream buf;
+
+  PrintTimerDescription(&buf);
+  Timer timer(&buf);
+  timer.Start();
+
+  usleep(usleep_time);
```

Comment:
This kind of tests are usually done through mocking. For example you make a MockTimer class where functions return values which you define in the test.

Reply:
Done.